engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/source/BaseKafkaAvroSourceFactory.scala

```diff
@@ -24,23 +24,34 @@ abstract class BaseKafkaAvroSourceFactory[T: ClassTag](timestampAssigner: Option
                    kafkaConfig: KafkaConfig,
                    deserializationSchemaFactory: KafkaAvroDeserializationSchemaFactory,
                    createRecordFormatter: String => Option[RecordFormatter],
-                   schemaDeterminer: AvroSchemaDeterminer)
+                   schemaDeterminer: AvroSchemaDeterminer,
+                   returnGenericAvroType: Boolean)
                   (implicit processMetaData: MetaData,
-                   nodeId: NodeId): KafkaSource[T] with ReturningType = {
+                   nodeId: NodeId): KafkaSource[T] = {
 
     val schema = schemaDeterminer.determineSchemaUsedInTyping.valueOr(SchemaDeterminerErrorHandler.handleSchemaRegistryErrorAndThrowException)
-    val returnTypeDefinition = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema)
     val schemaUsedInRuntime = schemaDeterminer.toRuntimeSchema(schema)
 
-    new KafkaSource(
-      List(preparedTopic),
-      kafkaConfig,
-      deserializationSchemaFactory.create[T](schemaUsedInRuntime, kafkaConfig),
-      assignerToUse(kafkaConfig),
-      createRecordFormatter(preparedTopic.prepared),
-      TestParsingUtils.newLineSplit
-    ) with ReturningType {
-      override def returnType: typing.TypingResult = returnTypeDefinition
+    if (returnGenericAvroType) {
```

Comment:
Mayby shorter:

````
override def returnType: typing.TypingResult = {
if (returnGenericAvroType) {
AvroSchemaTypeDefinitionExtractor.typeDefinition(schema)
} else {
super.returnType
}
}

Reply:
It won't work, because `ReturningType` is a marker interface and it will change a behaviour